ArjunaJTA/jta/classes/com/arjuna/ats/internal/jta/resources/arjunacore/XAResourceRecord.java

```diff
@@ -318,6 +322,7 @@ public int topLevelAbort()
 				    }
 				    else
 				    {
+						// TODO: Is this in the right catch block?
```

Comment:
@jmfinelli please can you elaborate on the TODO? Does it need resolving?

Reply:
This is not the best of the explanation, isn't it :sweat_smile: ?

What I wanted to discuss here is:
```
Why is ArjunaJTA returning the message with ID 16045
(i.e. "attempted rollback of {0} ({1}) failed with exception code {2}")
when we're not catching error codes from a rollback attempt?
```
Here, we're catching exceptions generated by an `end` attempt. Although we're passing TMFAIL as parameter, AFAIU, the XA specs says that:
```
A resource manager might choose to mark a transaction branch
as rollback-only at this point. In fact, a transaction manager does so
for the global transaction. If a resource manager chooses to do so also,
xa_end() returns one of the [XA_RB∗] values.
```
i.e. ArjunaJTA should call rollback on that RM itself when the previous `end` invocation fails. Thus, It seems to me that the two messages with ID 160450 shouldn't be used here (i.e. line 326 and 337) Please, could you help me understand this part of the code? @tomjenkinson

Reply:
I will take a look into this but I think it will be tomorrow at least now - thanks for asking

Reply:
Regarding the `else` - did you consider the >= XA_RBBASE in the first catch? (https://github.com/jmfinelli/narayana/blob/JBTM-3843/ArjunaJTA/jta/classes/com/arjuna/ats/internal/jta/resources/arjunacore/XAResourceRecord.java#L263)?

Reply:
Yep, the subsequent rollback [invocation](https://github.com/jmfinelli/narayana/blob/JBTM-3843/ArjunaJTA/jta/classes/com/arjuna/ats/internal/jta/resources/arjunacore/XAResourceRecord.java#L305) should cover that case

Reply:
I think I understand from the other thread that you are concerned it's an xa_end call rather than a rollback and so rather than an XA problem or additional logging, you are thinking about the message talking about rollback rather than xa_end. As the logging is within a call regarding abort, personally I expect I would leave it as (if there was no connection to rollback at this point I hope I would feel differently).

Reply:
OK, I'm gonna resolve this discussion. Thanks Tom